### ISSUE TITLE
terraform plans to update NR resources which are not updateable 

### ISSUE THREAD

**Issue by rblumen-desk** [terraform-providers/terraform-provider-newrelic#390]
### Terraform Version
Terraform v0.12.20
+ provider.newrelic v1.13.0

### Affected Resource(s)
Please list the resources as a list, for example:
newrelic_alert_channel
newrelic_alert_policy_channel shows similar issues

### Terraform Configuration Files
```hcl
terraform {
  required_version = "0.12.20"

  backend "local" {
    path = "tfstate"
  }
}

provider newrelic {
  version = "1.13.0"
}

resource newrelic_alert_channel test-config {
  name = "slack channel 1 for testing"
  type = "slack"

  config {
    channel = "#dev-null"
    url     = "https://hooks.slack.com/services/<redacted>/<redacted>"
  }
}

resource newrelic_alert_channel test-configuration {
  name = "slack channel 2 for testing"
  type = "slack"

  configuration = {
    channel = "#dev-null"
    url     = "https://hooks.slack.com/services/<redacted>/<redacted>"
  }
}

```

### Debug Output
https://gist.github.com/rblumen-desk/25ea23d67ddfbb3c6a24f39af1e995b4

### Expected Behavior
The terraform code shows two resources.  They are slightly different configurations of the newrelic_alert_channel.  The alert channel resource [does not provide an Update function|https://github.com/terraform-providers/terraform-provider-newrelic/blob/a7b4deae08415da3e92316c40ad54e0a59c1b9b6/newrelic/resource_newrelic_alert_channel.go#L56] and is not updateable in place.   

1. First time through terraform plan shows that both resources will be created.
2. First time through terraform apply creates the resources.
3. Running plan again, I expect that it should show no changes required at all.
4. Running apply again should do nothing.

### Actual Behavior

Running plan a second time shows that the channel configured with "configuration" needs to be replaced and the resource configured with "config" needs to be update in place - even though this resource type does not support update in-place:

3. Second time through, running plan again with *no changes* shows that the first resource, configured with a "config" block will be updated in place while the second resource will be replaced. 
4.  Second time through, running apply fails because the resource does not support update in place.  The plan that is generated is not possible to apply. 

plan:
```
Terraform will perform the following actions:

  # newrelic_alert_channel.test-config will be updated in-place
  ~ resource "newrelic_alert_channel" "test-config" {
        id   = "3161158"
        name = "slack channel 1 for testing"
        type = "slack"
      ~ config {
            channel = "#scrt-dev-null"
            headers = (sensitive value)
            payload = (sensitive value)
          + url     = (sensitive value)
        }
    }
  # newrelic_alert_channel.test-configuration must be replaced
-/+ resource "newrelic_alert_channel" "test-configuration" {
      ~ configuration = (sensitive value)
      ~ id            = "3161157" -> (known after apply)
        name          = "slack channel 2 for testing"
        type          = "slack"
    }
Plan: 1 to add, 1 to change, 1 to destroy.
Warning: "configuration": [DEPRECATED] use `config` block instead

  on main.tf line 23, in resource "newrelic_alert_channel" "test-configuration":
  23: resource newrelic_alert_channel test-configuration {
```

Running apply a second time fails because the resource can not be updated in place:

```
newrelic_alert_channel.test-configuration: Refreshing state... [id=3161190]
newrelic_alert_channel.test-config: Refreshing state... [id=3161158]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # newrelic_alert_channel.test-config will be updated in-place
  ~ resource "newrelic_alert_channel" "test-config" {
        id   = "3161158"
        name = "slack channel 1 for testing"
        type = "slack"

      ~ config {
            channel = "#scrt-dev-null"
            headers = (sensitive value)
            payload = (sensitive value)
          + url     = (sensitive value)
        }
    }

  # newrelic_alert_channel.test-configuration must be replaced
-/+ resource "newrelic_alert_channel" "test-configuration" {
      ~ configuration = (sensitive value)
      ~ id            = "3161190" -> (known after apply)
        name          = "slack channel 2 for testing"
        type          = "slack"
    }

Plan: 1 to add, 1 to change, 1 to destroy.


Warning: "configuration": [DEPRECATED] use `config` block instead

  on main.tf line 23, in resource "newrelic_alert_channel" "test-configuration":
  23: resource newrelic_alert_channel test-configuration {

newrelic_alert_channel.test-configuration: Destroying... [id=3161190]
newrelic_alert_channel.test-config: Modifying... [id=3161158]
newrelic_alert_channel.test-configuration: Destruction complete after 1s
newrelic_alert_channel.test-configuration: Creating...
newrelic_alert_channel.test-configuration: Creation complete after 0s [id=3161197]

Error: doesn't support update
```
### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` once
2. `terraform apply` a second time 

### Important Factoids
To execute these scripts requires a terraform account and a slack account and the corresponding API keys.  I have observed this issue with all flavors of newrelic_alert_channel so there is nothing special about the slack variant. 

### References
I do not see any on this provider.  This could be an engine issue.

**Comment by zlesnr** [comment]
Thank you for the report.  I believe this was fixed in https://github.com/terraform-providers/terraform-provider-newrelic/pull/387 but we have not cut a release yet.  Likely Wednesday the new release will include the fix here.

**Comment by w32blaster** [comment]
hey, we also faced with this issue :( I use provider "newelic" version 1.14.0

**Comment by rblumen-desk** [comment]
The issue was fixed for me with 13.1.

**Comment by zlesnr** [comment]
@w32blaster Would you mind filing a new issue if you're still having trouble here.  I believe the issue here is resolved.
